dist/svg.js

```diff
@@ -1694,20 +1704,53 @@ SVG.FX = SVG.invent({
 
       // convert current time to position
       if(!ignoreTime) this.pos = this.timeToPos(+new Date)
+      else {
+        // If the method at was called with a position above 1, we set it back to 1
+        if(this.pos > 1) this.pos = 1
+        if(this.situation.reversed) this.pos = 1 - this.pos
+      }
 
-      if(this.pos >= 1 && (this.situation.loop === true || (typeof this.situation.loop == 'number' && --this.situation.loop))){
+      if(this.pos < 0) this.pos = 0
+      if(this.pos >= 1){
+        // It can happen that position is many times above 1, especially when the duration is really small.
+        // That mean that step is not called fast enough to update the animation. To compensate, we use
+        // the integer part of the position to adjust the loop counter and decide if reversed should
+        // be toggled or not.
+        if(this.situation.loop === true || (typeof this.situation.loop == 'number' && this.situation.loop)) {
+          var currentLoop = true
+            , lastLoop = this.situation.loop
+            , posInt = Math.floor(this.pos)
+
+          if(lastLoop !== true) currentLoop = Math.max(lastLoop-posInt, 0)
+          if(currentLoop) {
+            this.pos -= posInt
+            ignoreTime = true
+            if(this.situation.reversing)
+              // Toggle reversed if posInt is odd
+              this.situation.reversed = this.situation.reversed != (posInt % 2)
+          } else {
+            this.pos = 1
+            ignoreTime = false
+            if(this.situation.reversing)
+              // Toggle reversed if lastLoop-1 is odd
+              // The -1 here is because we don't want to toggle reversed when all the loops have been completed
+              this.situation.reversed = this.situation.reversed != ((lastLoop-1) % 2)
+          }
 
-        if(this.situation.reversing){
-          this.situation.reversed = !this.situation.reversed
+          this.situation.loop = currentLoop
+        } else {
+          this.pos = 1
+          ignoreTime = false
         }
-        return this.at(this.pos-1)
       }
 
-      if(this.situation.reversed) this.pos = 1 - this.pos
+      if(ignoreTime){
```

Comment:
I really overused this parameter XD! I have used it here because this part need to also be done when _step_ is called with _ignoreTime_ set to _true_. What I could do to make it more readable is to use a new boolean variable instead, something like _recalculateStart_.